### DISCUSSION TITLE
using alpine js with twig

### DISCUSSION THREAD

**Discussion by merijnponzo** [alpinejs/alpine#2071] [1. Help]
hello, how would i use twig with alpinejs

Since twig use {{ double curly brackets }} this button below

```
{% set twig_var = 0 %} 
<button x-on:click="next = {{twig_var}}">Toggle</button>
```
results into an error:
```
Uncaught SyntaxError: expected expression, got '}'
```

**Comment by SimoTod** [comment]
Isn't twig evaluated on the server? Alpine should never see the double brackets

**Comment by merijnponzo** [comment]
Yes twig is rendered on the server, the problem lies within the double curly brackets, since the example below is working properly

```
<?php $php_var= 0;?>
	<button x-on:click="next = <?php echo $php_var;?>">Toggle</button>
```

**Comment by merijnponzo** [comment]
I guess there was nothing wrong with the code but i had to define 'next' first ;-)

```
{% set twig_var = 222 %} 
<div x-data="{ next : 0 }">
<button x-on:click="next = {{twig_var}}">Toggle</button>
<span x-text="next">var</span>
</div>
```
